### PR TITLE
Updated TapToPlace.cs

### PR DESCRIPTION
**Pull request by StephenHodgson**
- Now works in both the editor and on device.
- No longer requires the spatial mapping manager.
- Now requires a `Collider` & `Interpolator` components.
- Recursively sets layer to `IgnoreRaycast` when placing the object and subscribes to a global click event for placement.
- Updated Tap To Place Test scene.
- Added a warning in `WorldAnchorStore` that the anchor store is never ready when in editor.

**Breaking Changes:**
- Updated the namespace to `HoloToolkit.Unity.InputModule` now that we're not dependent on the `SpatialMappingManager`.
- Removed the protected cached manager fields.
    - Devs should just reference these managers directly without the requirements of these fields in an inherited class.

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@jessemcculloch any chance of a review?

**Comment by StephenHodgson** [comment]
@jessemcculloch thanks! made a few more updates.

**Comment by StephenHodgson** [comment]
@thebanjomatic that's a good point.

File: Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            {
+                Debug.LogWarning("The specified parent object is not a parent of this object.");
+            }
+        }
+
+        private static void SetLayerRecursively(Transform objectToSet, bool useDefaultLayer)
+        {
+            if (useDefaultLayer)
+            {
+                if (defaultLayersCache.ContainsKey(objectToSet.gameObject))
```
Comment (thebanjomatic) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: should this be a loop over key-values instead (and the clearing the list after?

Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: Because there's no real way to keep track of any children that get added or removed.

Comment (thebanjomatic) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: The defaultLayersCache should be a collection of objects that had their layers changed when placement started. If new game objects are added after that, they don't need their values reset either.

Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: That's the way it works currently. I'm more worried about is we remove child objects.

Comment (thebanjomatic) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: Ah, nevermind. I think I thought the recrusive part was part of the else branch only. I do have one remaining comment here in that the ContainsKey check is unnecessary if using TryGetValue (which itself does the equivalent ContainsKey check).

Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: Fixed, Thanks for the review!

File: Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// The TapToPlace class is a basic way to enable users to move objects 
+    /// and place them on real world surfaces.
+    /// Put this script on the object you want to be able to move. 
+    /// Users will be able to tap objects, gaze elsewhere, and perform the
+    /// tap gesture again to place.
+    /// This script is used in conjunction with GazeManager, GestureManager,
```
Comment (thebanjomatic) [inline Assets/HoloToolkit/Input/Scripts/Interactions/TapToPlace.cs]: No longer uses GestureManager

**Pull request merged**